errors triggered in third party libraries

The recently added error (formerly warning) categories `register` and `ignored-attributes` trigger errors in pybind11:
```
In file included from /usr/include/python2.7/Python.h:88:0,
                 from /opt/ros/melodic/include/pybind11_catkin/pybind11/detail/common.h:112,
                 from /opt/ros/melodic/include/pybind11_catkin/pybind11/pytypes.h:12,
                 from /opt/ros/melodic/include/pybind11_catkin/pybind11/cast.h:13,
                 from /opt/ros/melodic/include/pybind11_catkin/pybind11/attr.h:13,
                 from /opt/ros/melodic/include/pybind11_catkin/pybind11/pybind11.h:43,
                 from [some file that includes pybind11/pybind11.h]:1:
/usr/include/python2.7/unicodeobject.h:534:24: error: ISO C++1z does not allow 'register' storage class specifier [-Werror=register]
     register PyObject *obj,     /* Object */
                        ^~~
/usr/include/python2.7/unicodeobject.h:553:24: error: ISO C++1z does not allow 'register' storage class specifier [-Werror=register]
     register PyObject *obj      /* Object */
                        ^~~
/usr/include/python2.7/unicodeobject.h:575:29: error: ISO C++1z does not allow 'register' storage class specifier [-Werror=register]
     register const wchar_t *w,  /* wchar_t buffer */
                             ^
/usr/include/python2.7/unicodeobject.h:593:23: error: ISO C++1z does not allow 'register' storage class specifier [-Werror=register]
     register wchar_t *w,        /* wchar_t buffer */
```

and Eigen:
```
In file included from /usr/include/eigen3/unsupported/Eigen/CXX11/Tensor:105:0,
                 from [exotica_ws]/src/exotica/exotica_core/include/exotica_core/tools/conversions.h:43,
                 from [exotica_ws]/src/exotica/exotica_core/include/exotica_core/tools.h:37,
                 from [exotica_ws]/src/exotica/exotica_core/include/exotica_core/object.h:33,
                 from [exotica_ws]/src/exotica/exotica_core/include/exotica_core/factory.h:37,
                 from [exotica_ws]/src/exotica/exotica_core/include/exotica_core/task_map.h:39,
                 from [...]:
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorEvaluator.h: In instantiation of 'struct Eigen::TensorEvaluator<const Eigen::TensorMap<Eigen::Tensor<float, 3, 1> >, Eigen::DefaultDevice>':
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorAssign.h:96:70:   required from 'struct Eigen::TensorEvaluator<const Eigen::TensorAssignOp<Eigen::Tensor<float, 3, 1>, const Eigen::TensorMap<Eigen::Tensor<float, 3, 1> > >, Eigen::DefaultDevice>'
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:397:14:   required from 'Eigen::Tensor<Scalar_, NumIndices_, Options_, IndexType>::Tensor(const Eigen::TensorBase<OtherDerived, 1>&) [with OtherDerived = Eigen::TensorMap<Eigen::Tensor<float, 3, 1> >; Scalar_ = float; int NumIndices_ = 3; int Options_ = 1; IndexType_ = long int]'
/usr/include/c++/7/bits/unique_ptr.h:825:30:   required from 'typename std::_MakeUniq<_Tp>::__single_object std::make_unique(_Args&& ...) [with _Tp = const Eigen::Tensor<float, 3, 1>; _Args = {Eigen::TensorMap<Eigen::Tensor<float, 3, 1, long int>, 0, Eigen::MakePointer>}; typename std::_MakeUniq<_Tp>::__single_object = std::unique_ptr<const Eigen::Tensor<float, 3, 1> >]'
[source file using Eigen::Tensor]:   required from here
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorEvaluator.h:162:71: error: ignoring attributes on template argument 'Eigen::PacketType<float, Eigen::DefaultDevice>::type {aka __vector(4) float}' [-Werror=ignored-attributes]
     PacketAccess = (internal::unpacket_traits<PacketReturnType>::size > 1),
```

I remember that I got the pybind11 warnings before.
Should we really turn those warnings into errors? I would wait until they are fixed upstream before we activate them in EXOTica.

--- a/exotica_core/cmake/exotica.cmake
+++ b/exotica_core/cmake/exotica.cmake
@@ -1,7 +1,8 @@
 cmake_minimum_required(VERSION 2.8.3)
 
 message(STATUS "Compiling using c++ 11 (required by EXOTica)")
-add_compile_options(-std=c++11 -Werror -Wfloat-conversion)
+add_compile_options(-std=c++11 -Wfloat-conversion)
+# add_compile_options(-Werror)
 
 # MoveIt Core Robot Model isnt aligned :'(
 #add_definitions(-DEIGEN_MAX_ALIGN_BYTES=0 -DEIGEN_DONT_VECTORIZE)

